libs/extensions/angular/sidebar-menu/src/public-components/sidebar-header/sidebar-header.component.ts

```diff
@@ -2,16 +2,47 @@ import { Component } from '@angular/core';
 
 @Component({
   selector: 'kirby-x-sidebar-header',
-  template: '<span class="sidebar-logo"><ng-content></ng-content></span>',
-  styles: [
-    `
-      .sidebar-logo {
-        display: flex;
-        align-items: center;
-        margin: 32px 0;
-        height: 50px;
+  template: `
+    <span class="sidebar-logo"><ng-content></ng-content></span>
+    <div class="action-bar">
+      <ng-content select="[slot='search']"></ng-content>
+      <ng-content select="[slot='action']"></ng-content>
+    </div>
+  `,
+  styles: `
+    :host {
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      flex-grow: 1;
+    }
+
+    :host::ng-deep .sidebar-logo > img {
+      object-fit: contain;
+      width: 100%;
+      height: 100%;
+    }
+
+    .sidebar-logo {
+      display: flex;
+      align-items: center;
+      height: 50px;
+      margin: var(--kirby-spacing-l) var(--kirby-spacing-m);
+    }
+
+    .action-bar {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      justify-content: center;
+      box-sizing: border-box;
+      width: 100%;
+
+      &:has(button) {
```

Comment:
If the logic is that padding should be adjused if anything is added to a slot, consider this:

```suggestion
      &:has([slot="search"]),
      &:has([slot="action"]) {
```

Reply:
done